packages/cpp-debug/src/browser/memory-provider/memory-provider.ts

```diff
@@ -14,22 +14,10 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { injectable, inject } from 'inversify';
+import { Interfaces } from '../utils/memory-widget-utils';
 import { DebugSessionManager } from '@theia/debug/lib/browser/debug-session-manager';
-
-/**
- * Representation of a memory read result.
- */
-export interface MemoryReadResult {
-    /**
-     * The bytes for the read result.
-     */
-    bytes: Uint8Array,
-    /**
-     * The address for the read result.
-     */
-    address: number,
-}
+import { injectable, inject } from 'inversify';
+import { hexStrToUnsignedLong } from '../../common/util';
 
 export const MemoryProvider = Symbol('MemoryProvider');
```

Comment:
@thegecko, this is the interface between the debugger backend and the frontend. The current implementation knows it's interacting with GDB because the CDT-GDB adapter doesn't implement memory reading and writing as prescribed by the debug adapter protocol, but it should be relatively straightforward to swap it out.

It might be desirable to implement something like a `MemoryProviderService` that gathers `MemoryProvider` implementations and matches a given debug session with the memory provider designed for it. Let us know if that sounds useful, and we'll be happy to add it.